content-security-policy/script-src/script-src-strict_dynamic_parser_inserted_correct_nonce.html

```diff
@@ -0,0 +1,86 @@
+
+    <script nonce='dummy'>
+        async_test(function(t) {
+            document.writeln('<scr' + 'ipt nonce="dummy" id="documentWriteln" src="simpleSourcedScript.js"></scr' + 'ipt>');
+            assert_not_equals(document.getElementById('documentWriteln'), null);
+            t.done();
+        }, 'Parser-inserted script via `document.writeln` with a correct nonce is allowed with `strict-dynamic`.');
+    </script>
+
+    <script nonce='dummy'>
+        async_test(function(t) {
+            document.write('<scr' + 'ipt defer nonce="dummy" id="documentWrite-defer" src="simpleSourcedScript.js"></scr' + 'ipt>');
+            assert_not_equals(document.getElementById('documentWrite-defer'), null);
+            t.done();
+        }, 'Parser-inserted deferred script via `document.write` with a correct nonce is allowed with `strict-dynamic`.');
+    </script>
+
+    <script nonce='dummy'>
+        async_test(function(t) {
+            document.writeln('<scr' + 'ipt defer nonce="dummy" id="documentWriteln-defer" src="simpleSourcedScript.js"></scr' + 'ipt>');
+            assert_not_equals(document.getElementById('documentWriteln-defer'), null);
+            t.done();
+        }, 'Parser-inserted deferred script via `document.writeln` with a correct nonce is allowed with `strict-dynamic`.');
+    </script>
+
+    <script nonce='dummy'>
+        async_test(function(t) {
+            document.write('<scr' + 'ipt async nonce="dummy" id="documentWrite-async" src="simpleSourcedScript.js"></scr' + 'ipt>');
+            assert_not_equals(document.getElementById('documentWrite-async'), null);
+            t.done();
+        }, 'Parser-inserted async script via `document.write` with a correct nonce is allowed with `strict-dynamic`.');
+    </script>
+
+    <script nonce='dummy'>
+        async_test(function(t) {
+            document.writeln('<scr' + 'ipt async nonce="dummy" id="documentWriteln-async" src="simpleSourcedScript.js"></scr' + 'ipt>');
+            assert_not_equals(document.getElementById('documentWriteln-async'), null);
+            t.done();
+        }, 'Parser-inserted async script via `document.writeln` with a correct nonce is allowed with `strict-dynamic`.');
+    </script>
+
+    <script nonce='dummy'>
+        async_test(function(t) {
+            document.write('<scr' + 'ipt defer async nonce="dummy" id="documentWrite-defer-async" src="simpleSourcedScript.js"></scr' + 'ipt>');
+            assert_not_equals(document.getElementById('documentWrite-defer-async'), null);
+            t.done();
+        }, 'Parser-inserted deferred async script via `document.write` with a correct nonce is allowed with `strict-dynamic`.');
+    </script>
+
+    <script nonce='dummy'>
+        async_test(function(t) {
+            document.writeln('<scr' + 'ipt defer async nonce="dummy" id="documentWriteln-defer-async" src="simpleSourcedScript.js"></scr' + 'ipt>');
+            assert_not_equals(document.getElementById('documentWriteln-defer-async'), null);
```

Comment:
These are only checking that the element exists, right? Shouldn't you be checking at some point that the relevant element has a value set on it? Perhaps another approach would be to have a script that posts a message to itself: something like:

```
// amazing-script-name.js
window.postMessage(document.currentScript.id, "*");
```

and

```
// amazing test
async_text(function (t) {
  window.addEventListener("message", t.step_func(function (e) {
    if (e.data == "element-id-from-below")
      t.done();
  }));
  document.write("<script id='element-id-from-below' ...></scr" + "ipt>");
}, "Testy test.");
```

WDYT?